Custom metrics (#6213)
DataDog/documentation: content/en/developers/metrics/_index.md

  text: "Official and Community created API and DogStatsD client libraries"
---

## Overview

> If a metric is not submitted from one of the [350+ Datadog integrations][1] it's considered a custom metric<sup>(1)</sup>.

Reviewer: Can we add something about what a custom metric is, rather than strictly what it is not, or a link to a neat blog? Custom metrics measure often telemetry about an application, Examples of custom metrics include critical business telemetry like time on page, or other operational data about the behavior of a custom application or something?

A custom metric is identified by **a unique combination of a metric's name and tag values (including the host tag)**. In general, any metric you send using [DogStatsD][2] or through a [custom Agent Check][3] is a custom metric. Custom metrics help you track your application KPIs: number of visitors, average customer basket size, request latency, or performance distribution for a custom algorithm.

> **Note**: User with the Datadog Admin roles can see the total custom metrics per hour and the top 500 custom metrics for their account in the [usage details page][4]. Learn more about [how custom metrics are counted][5].

Reviewer 2: ```suggestion
**Note**: Users with the Datadog Admin roles can see the total custom metrics per hour and the top 500 custom metrics for their account in the [usage details page][4]. Learn more about [how custom metrics are counted][5].
```

> ## Custom Metric properties

Reviewer 2: ```suggestion
## Custom metrics properties
```

> If a metric is not submitted from one of the [350+ Datadog integrations][1] it's considered a custom metric<sup>(1)</sup>. Custom metrics help you track your application KPIs: number of visitors, average customer basket size, request latency, or performance distribution for a custom algorithm.

Reviewer 3: sup link is broken

> If a metric is not submitted from one of the [350+ Datadog integrations][1] it's considered a custom metric<sup>[(1)][2]</sup>. Custom metrics help you track your application KPIs: number of visitors, average customer basket size, request latency, or performance distribution for a custom algorithm.

Reviewer 3: sup still broken

Author: It's working for me.. could you record what you are seeing ?

> A custom metric is identified by **a unique combination of a metric's name and tag values (including the host tag)**. In general, any metric you send using [DogStatsD][2] or through a [custom Agent Check][3] is a custom metric.

Reviewer 3: A custom metric is identified by **a unique combination of a metric name and tag values (including the host tag)**. In general, any metric you send using [DogStatsD][2] or through a [custom Agent Check][3] is a custom metric.

> **Note**: Users with the Datadog Admin roles can see the total custom metrics per hour and the top 500 custom metrics for their account in the [usage details page][4]. Learn more about [how custom metrics are counted][5].

Reviewer 3: **Note**: Users with the Datadog Admin roles can see the monthly average number of custom metrics per hour and the top 500 custom metrics for their account in the [usage details page][4]. Learn more about [how custom metrics are counted][5].

A Datadog custom metric is defined by the properties below. Refer to the [Metrics Introduction documentation][6] to learn how to graph metrics within Datadog.

| Property         | Description                                                                                                                                               |
|------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------|
| `<METRIC_NAME>`  | The [name of your metric](#naming-metrics).                                                                                                               |
| `<METRIC_VALUE>` | The value of your metric.                                                                                                                                 |

> | `<TIMESTAMP>`    | The timestamp associated with the metric value. **Note**: Metric timestamps cannot be more than 10 minutes in the future or more than 1 hour in the past. |

Reviewer 2: ```suggestion
| `<TIMESTAMP>`    | The timestamp associated with the metric value. **Note**: Metric timestamps cannot be more than ten minutes in the future or more than one hour in the past. |
```

| `<TAGS>`         | The set of tags associated with your metric.                                                                                                              |
| `<METRIC_TYPE>`  | The type of your metric. See the [metric type documentation][7].                                                                                          |
| `<INTERVAL>`     | If the `<TYPE>` of the metric is [RATE][8] or [COUNT][9], it defines the corresponding [interval][10].                                                    |

### Naming custom metrics

The following custom metric naming convention must be followed:

> * Metric names must start with a letter

Reviewer 2: ```suggestion
* Metric names must start with a letter.
```

    {{< nextlink href="/api/?lang=python#post-timeseries-points" >}}Datadog's HTTP API{{< /nextlink >}}
{{< /whatsnext >}}

> You can also use one of the [Datadog-official and community contributed API and DogStatsD client libraries][11] to submit your custom metrics

Reviewer 2: ```suggestion
You can also use one of the [Datadog official and community contributed API and DogStatsD client libraries][11] to submit your custom metrics
```
